test/runtests.jl

```diff
@@ -12,7 +12,18 @@ else
 end
 
 if VERSION >= v"0.5.0"
-    @test isempty(detect_ambiguities(ColorTypes, Base, Core))
+    if isdefined(Core, :UnionAll)
```

Comment:
ATM I think this branch is really about skipping the ambiguity test, which does coincide with `UnionAll`. When we uncomment that line, then I guess we have to be more careful.